Rakefile

```diff
@@ -1,9 +1,102 @@
 # Add your own tasks in files placed in lib/tasks ending in .rake,
 # for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
 
+require "json"
+require "net/http"
+require "shellwords"
+
 require_relative "config/application"
 
 Rails.application.load_tasks
 
 Rake::Task[:default].clear if Rake::Task.task_defined?(:default)
 task default: %i[lint test]
+
+desc "Push the commit at the tip of the current branch to the dev environment"
+task push_to_dev: :environment do
+  puts "Getting Git commit..."
+  git_hash = ""
+  run "git", "rev-parse", "HEAD" do |ok, value, _|
+    ok or raise
+    git_hash = value.strip
+  end
+  puts "Will use Git commit: #{git_hash}"
+  puts "Checking commit has been pushed..."
+
+  resp = Net::HTTP.get_response URI("https://api.github.com/repos/alphagov/forms-product-page/commits/#{git_hash}")
+  if resp.code != "200"
+    puts "Commit has not been pushed. The pipelines will not be able to find it."
+    puts "Push the commit and run the command again."
+    raise
+  end
+  puts "The commit is present in the remote"
+
+  push_commit_to_dev(git_hash)
+end
+
+desc "Push the commit at the tip of origin/main to the dev environment"
+task reset_dev_to_main: :environment do
+  puts "Updating Git remotes"
+  sh "git", "fetch", verbose: false
+
+  git_hash = ""
+  run "git", "rev-parse", "origin/main" do |ok, value, _|
+    ok or raise
+    git_hash = value.strip
+  end
+
+  puts "Will use Git commit from tip of origin/main: #{git_hash}"
+
+  push_commit_to_dev(git_hash)
+end
+
+def push_commit_to_dev(git_hash)
+  json_input = {
+    name: "forms-product-page-image-builder",
+    sourceRevisions: [
+      {
+        actionName: "get-forms-product-page",
+        revisionType: "COMMIT_ID",
+        revisionValue: git_hash,
+      },
+    ],
+    variables: [
+      {
+        name: "tag_prefix",
+        value: "dev-",
+      },
+    ],
+  }
+
+  puts ""
+  puts "Starting pipeline 'forms-product-page-image-builder' in the 'deploy' account"
+  pp json_input
+
+  cli_input = JSON.dump(json_input)
+  command_args = [
+    "aws",
+    "codepipeline",
+    "start-pipeline-execution",
+    "--name",
+    "forms-product-page-image-builder",
+    "--cli-input-json",
+    cli_input,
+  ]
+  sh_aws(*command_args)
+end
+
+def sh_aws(*cmd)
+  if ENV.key?("AWS_VAULT") && (ENV["AWS_VAULT"] != "forms-deploy-support")
+    ENV.delete("AWS_VAULT")
+  end
+
+  all_args = ["gds", "aws", "forms-deploy-support", "--", *cmd]
+  system(*all_args)
```

Comment:
In this instance I wanted to maintain the entire rest of the environment, aside from the `AWS_VAULT` key if it had the wrong value. We do this because you can't have nested AWS Vault sessions, which are detected by the presence and value of the `AWS_VAULT` env var.

I guess I could take a copy of the `ENV` hash, remove the key as necessary, and pass that to `system`?

Reply:
Have gone with your suggestion. You know Ruby and its idioms better than I